core/src/main/scala/cats/std/either.scala

```diff
@@ -23,7 +23,7 @@ trait EitherInstances extends EitherInstances1 {
         }
     }
 
-  implicit def eitherInstances[A]: Monad[Either[A, ?]] with Traverse[Either[A, ?]] =
+  implicit def catsMonadForEither[A]: Monad[Either[A, ?]] with Traverse[Either[A, ?]] =
```

Comment:
There's probably no single good solution here, but I think that including _all_ of the type class names in the `implicit` name has its issues too. Changing the name of these `implicits` breaks binary compatibility. Ideally we could make an `implicit` extend another type class without breaking binary compatibility but then we wouldn't have the new type class in the name.